ledger/catchpointtracker.go

```diff
@@ -1446,21 +1442,55 @@ func accountHashBuilderV6(addr basics.Address, accountData *baseAccountData, enc
 	return finishV6(hash, prehash)
 }
 
-// accountHashBuilderV6 calculates the hash key used for the trie by combining the account address and the account data
-func resourcesHashBuilderV6(addr basics.Address, cidx basics.CreatableIndex, ctype basics.CreatableType, updateRound uint64, encodedResourceData []byte) []byte {
-	hash := hashBufV6(updateRound, byte(ctype+1)) // one or two ( asset / application ) so we could differentiate the hashes.
+// hashKind enumerates the possible data types hashed into a catchpoint merkle
+// trie.  Each merkle trie hash includes the hashKind byte at a known-offset.
+// By encoding hashKind at a known-offset, it's possible for hash readers to
+// disambiguate the hashed resource.
+//go:generate stringer -type=hashKind
+type hashKind byte
+
+// Defines known kinds of hashes.  Changing an enum ordinal value is a
+// breaking change.
+const (
+	accountHK hashKind = iota
+	assetHK
+	appHK
+	kvHK
+)
+
+// HashKindEncodingIndex defines the []byte offset where the hash kind is
+// encoded.
+const HashKindEncodingIndex = 4
+
+func creatableHashKindFromResourcesData(rd resourcesData, a basics.Address, ci basics.CreatableIndex) (hashKind, error) {
```

Comment:
suggest a shorter name rdGetHashKind(...)

Reply:
@algorandskiy I refactored to a shorter name in https://github.com/algorand/go-algorand/pull/4808/commits/5fd57700d86ed8e23dc6b97ed0f9926559e86a5f.  Can you resolve if it looks good?